Lesson 6 - Developer tools/code/server.js

```diff
@@ -16,11 +21,31 @@ app.get('/today', function (req, res) {
 });
 
 //POST /numbers should save number for later
+app.post('/numbers', function (req, res) {
+    numbers.push(req.body);
+    res.send("Number saved for later");           
+});
 //GET /numbers should return comma separated list of numbers
+app.get('/numbers', function (req, res) {
+    setTimeout(function () {
+        if(numbers.length === 0){
+            res.send("empty");
+        }else{
+            res.send(numbers.join(", "));
+        }
+    }, 1000);
+});
 //DELETE /numbers should empty numbers array
+app.delete('/numbers', function (req, res) {
+    numbers.length = 0;
+    res.send("Number's array is empty");
+});
 //GET /numbers/:op should return result of given operation of stored numbers (check request.params)
 // available operations are sum, difference, multiplication, division
 // return result as strings
+app.get('/numbers/:op', function (req, res) {
+    
```

Comment:
I have problem with sum and difference, I have no idea why :(

https://github.com/michalek94/Kurs_UL

multiplication and division - works

2016-03-25 10:15 GMT+01:00 Rafał Warzycha <notifications@github.com>:

> In Lesson 6 - Developer tools/code/server.js
> <https://github.com/leszczynski-it/Kurs_UL/pull/8#discussion_r57429955>:
>
> >  //GET /numbers/:op should return result of given operation of stored numbers (check request.params)
> >  // available operations are sum, difference, multiplication, division
> >  // return result as strings
> > +app.get('/numbers/:op', function (req, res) {
> > +
>
> Here you need to implement calculator functionality based on numbers
> variable content. For each element in numbers, you need to use apply
> operation passed via parameter.
>
> In that function you will get op - operation parameter for in req
> variable - req.params.op. This /numbers/:op means that sample server call
> /number/sum or /number/difference and so forth.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/leszczynski-it/Kurs_UL/pull/8/files/39867fe3a81f6ad1c3094ffe8f006a25a262e073#r57429955>
>



-- 
Pozdrawiam,
Michał Pankowski.